scripts/zones/Lower_Delkfutts_Tower/Zone.lua

```diff
@@ -21,19 +21,48 @@ end
 
 function onZoneIn(player, prevZone)
     local cs = -1
+    local params = {0, 0, 0, 0, 0, 0, 0, 0}
 
-    if player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0 then
+    if
+        player:getXPos() == 0 and
+        player:getYPos() == 0 and
+        player:getZPos() == 0
+    then
         player:setPos(460.022, -1.77, -103.442, 188)
     end
-    if player:getCurrentMission(ZILART) == tpz.mission.id.zilart.RETURN_TO_DELKFUTTS_TOWER and player:getCharVar("ZilartStatus") <= 1 then
+
+    -- RETURN TO DELKFUTTS TOWER
+    if
+        player:getCurrentMission(ZILART) == tpz.mission.id.zilart.RETURN_TO_DELKFUTTS_TOWER and
+        player:getCharVar("ZilartStatus") <= 1
+    then
         cs = 15
-    elseif ENABLE_COP == 1 and prevZone == tpz.zone.QUFIM_ISLAND and player:getCurrentMission(COP) < tpz.mission.id.cop.THE_RITES_OF_LIFE then
+
+    -- THE RITES OF LIFE
+    elseif
+        ENABLE_COP == 1 and
+        prevZone == tpz.zone.QUFIM_ISLAND and
+        player:getCurrentMission(COP) < tpz.mission.id.cop.THE_RITES_OF_LIFE
+    then
         cs = 22
-    elseif player:getCurrentMission(ACP) == tpz.mission.id.acp.BORN_OF_HER_NIGHTMARES and prevZone == tpz.zone.QUFIM_ISLAND then
+
+    -- SPIRITS AWOKEN
+    elseif player:getCurrentMission(ROV) == tpz.mission.id.rov.SPIRITS_AWOKEN then
+        cs = 51
+
+        if player:getCurrentMission(COP) >= tpz.mission.id.cop.THREE_PATHS then
+            params[2] = 1 -- Acknowledges the Player Character already knowing Prishe
+        end
+
+    -- BORN OF HER NIGHTMARES
+    elseif
+        player:getCurrentMission(ACP) == tpz.mission.id.acp.BORN_OF_HER_NIGHTMARES and
+        prevZone == tpz.zone.QUFIM_ISLAND
+    then
         cs = 34
     end
 
-    return cs
+    return cs --,{params} need to pass a param for CS 51 to show extra dialog
```

Comment:
These params aren't passed as part of the cs start.
They're passed as a response to an EventUpdate the client sends with option 1. That response and option needs to be coded in onEvent Update below.

Reply:
:penguin:  ty